src/PythNetworkHTTPClient.ts

```diff
@@ -0,0 +1,153 @@
+
+    }
+
+    private handleProductAccount(key: PublicKey, account: AccountInfo<Buffer>) {
+        const {priceAccountKey, type, product} = parseProductData(account.data)
+        this.productAccountKeyToProduct[key.toString()] = product
+        if (priceAccountKey.toString() !== ONES) {
+            this.priceAccountKeyToProductAccountKey[priceAccountKey.toString()] = key.toString()
+        }
+
+        this._asset_types.add(product.asset_type);
+        this._product_symbols.add(product.symbol);
+        this._products.add(product);
+    }
+
+    private handlePriceAccount(key: PublicKey, account: AccountInfo<Buffer>) {
+        const product = this.productAccountKeyToProduct[this.priceAccountKeyToProductAccountKey[key.toString()]]
+        if (product === undefined) {
+            // This shouldn't happen since we're subscribed to all of the program's accounts,
+            // but let's be good defensive programmers.
+            throw new Error('Got a price update for an unknown product. This is a bug in the library, please report it to the developers.')
+        }
+
+        const priceData = parsePriceData(account.data)     
+        this._product_price.set(product, priceData);
+    }    
+
+    private handleAccount(key: PublicKey, account: AccountInfo<Buffer>, productOnly: boolean) {
+        const base = parseBaseData(account.data)
+        // The pyth program owns accounts that don't contain pyth data, which we can safely ignore.
+        if (base) {
+            switch (AccountType[base.type]) {
+                case 'Mapping':
+                    // We can skip these because we're going to get every account owned by this program anyway.
+                    break;
+                case 'Product':
+                    this.handleProductAccount(key, account)
+                    break;
+                case 'Price':
+                    if (!productOnly) {
+                        this.handlePriceAccount(key, account)
+                    } else {
+                        this._price_queue.push({
+                            key: key,
+                            account: account
+                        });
+                    }
+                    break;
+                case 'Test':
+                    break;
+                default:
+                    throw new Error(`Unknown account type: ${base.type}. Try upgrading pyth-client.`)
+            }
+        }
+    }
+
+    public async refreshData() {
+        const require_https = this._cluster === 'mainnet-beta' ? true : false;
+        const current_connection = new Connection(clusterApiUrl(this._cluster, require_https));
+
+        const accounts = await current_connection.getProgramAccounts(this._program_key, this._commitment);
+        for(const account of accounts) {
+            this.handleAccount(account.pubkey, account.account, true)
+        }
+
+        for(const queued of this._price_queue) {
+            this.handleAccount(queued.key, queued.account, false)
```

Comment:
can't you call `handlePriceAccount` directly here? I think that simplifies the logic, and also lets you get rid of the `productOnly` flag on `handleAccount`